swan-cern/scripts/10_spark.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Author: Danilo Piparo, Enric Tejedor, Pedro Maximino 2023
+# Copyright CERN
+# The environment for Spark is configured here.
+
+# Spark configuration
+if [[ $SPARK_CLUSTER_NAME ]]
+then
+  LOCAL_IP=`hostname -i`
+  echo "$LOCAL_IP $SERVER_HOSTNAME" >> /etc/hosts
+
+  # Enable the extensions in Jupyter global path to avoid having to maintain this information 
+  # in the user scratch json file (specially because now we persist this file in the user directory and
+  # we don't want to persist the Spark extensions across sessions)
+  mkdir -p /etc/jupyter/nbconfig
+  log_info "Globally enabling the Spark extensions"
+  echo "{
+    \"load_extensions\": {
+      \"sparkconnector/extension\": true,
+      \"hdfsbrowser/extension\": true
+    }
+  }" > /etc/jupyter/nbconfig/notebook.json
+  echo "{
+    \"NotebookApp\": {
+      \"nbserver_extensions\": {
+        \"hdfsbrowser.serverextension\": true
+      }
+    }
+  }" > /etc/jupyter/jupyter_notebook_config.json
+  if [ $SPARK_CLUSTER_NAME = "k8s" ]
+  then
+    NAMESPACE="analytix"
+    CLUSTER_NAME="analytix"
+  else
+    NAMESPACE=$(cat /cvmfs/sft.cern.ch/lcg/etc/hadoop-confext/conf/etc/$SPARK_CLUSTER_NAME/$SPARK_CLUSTER_NAME.info.json | python -c 'import json,sys;obj=json.load(sys.stdin);print(obj["namespace"])')
+    CLUSTER_NAME=$SPARK_CLUSTER_NAME
+  fi
+  echo "c.HDFSBrowserConfig.hdfs_site_path = '/cvmfs/sft.cern.ch/lcg/etc/hadoop-confext/conf/etc/$CLUSTER_NAME/hadoop.$CLUSTER_NAME/hdfs-site.xml'" >> $JPY_CONFIG
```

Comment:
Can we do a multiline print? It would be cleaner. I thought about putting this into a py file, but maybe it would be complicated given the env vars needed.

Reply:
We leave it with echo's, but just one multiline echo.

Reply:
Added a multi line echo.